### PR TITLE
Add variable scope and oh-context features to expression page

### PR DESCRIPTION
**Pull request by JustinGeorgi**
The new `oh-context` component includes several features important for advanced widget creation.  This PR adds descriptions and some examples of these features to the widget variables and expression page.

### CODE REVIEW

File: ui/widget-expressions-variables.md [hunk 1]
```diff
@@ -52,6 +52,64 @@ Variables can be used in several ways:
 `oh-button` & `oh-link` have a special parameter `clearVariable`, which allows to unset a variable when clicked, after performing the action.
 This is useful when "validating" a variable, e.g. send the variable value as command to an Item and then reset the variable.
 
+### Default Variable Values
+
+The standard variables defined using interactive components do exist until the first time they are given a value by a component.
```
Comment (florian-h05) [inline ui/widget-expressions-variables.md:57]: Missing word?

```suggestion
The standard variables defined using interactive components do not exist until the first time they are given a value by a component.
```

File: ui/widget-expressions-variables.md [hunk 2]
```diff
@@ -52,6 +52,64 @@ Variables can be used in several ways:
...
+        user: Guest
+        color: red
+        timeout: 5
+```
+
+### Variable Scope
+
+Standard variables are also global in scope with one exception:
+
+> the value of variable defined within one custom widget will not be available to the parent page or widget that custom widget is included in.
```
Comment (florian-h05) [inline ui/widget-expressions-variables.md:105]: Did you mean to say the following?

```suggestion
> Changes to the variable's value made by the child components do not affect the parent component that defines the variable.
```

File: ui/widget-expressions-variables.md [hunk 3]
```diff
@@ -52,6 +52,64 @@ Variables can be used in several ways:
...
+
+The scope of variables created using an `oh-context` component is restricted to only children of that component.
```
Comment (florian-h05) [inline ui/widget-expressions-variables.md:107]:
```suggestion
The scope of variables created using an `oh-context` component is restricted to only the children of that component.
```

File: ui/widget-expressions-variables.md [hunk 4]
```diff
@@ -199,6 +257,49 @@ Adding a simple `OR` statement to that expression will provide the fallback cond
...
+        config:
+          text: =@@hvacModeItem
+          style:
+            background: =const.modeColor[@@hvacModeItem] || 'red'
+```
+
+### Custom Functions
+
+There are times when a widget requires the same calculation in multiple locations and maintaining all the different locations can be a burden.
+In these instances, an `oh-context` component can be used to define named functions that are avialable to all children of the `oh-context`.
```
Comment (florian-h05) [inline ui/widget-expressions-variables.md:288]: Typo:

```suggestion
In these instances, an `oh-context` component can be used to define named functions that are available to all children of the `oh-context`.
```

**Review by florian-h05 — COMMENTED** [review]
Overall LGTM, thanks!

Just a few minor commens:

**Comment by JustinGeorgi** [comment]
Great edits, thanks. Should be all set now.

**Review by florian-h05 — APPROVED** [review]
LGTM, thanks!
Markdownlint fails unrelated.

@openhab/documentation-maintainers Can you please merge?

**Review by stefan-hoehn — APPROVED** [review]

**Pull request merged**